crates/ruma-common/src/identifiers/crypto_algorithms.rs

```diff
@@ -59,6 +59,40 @@ pub enum EventEncryptionAlgorithm {
     _Custom(PrivOwnedStr),
 }
 
+/// A key algorithm to be used to generate a key from a passphrase.
+///
+/// This type can hold an arbitrary string. To check for algorithms that are not available as a
+/// documented variant here, use its string representation, obtained through `.as_str()`.
```

Comment:
We have deduplicated this repeated doc string. Please replace with

```suggestion
#[doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/src/doc/string_enum.md"))]
```